server/src/main/java/org/elasticsearch/index/IndexSortConfig.java

```diff
@@ -112,12 +112,31 @@ private static MultiValueMode parseMultiValueMode(String value) {
     final FieldSortSpec[] sortSpecs;
     private final Version indexCreatedVersion;
     private final String indexName;
+    private final IndexMode indexMode;
 
     public IndexSortConfig(IndexSettings indexSettings) {
         final Settings settings = indexSettings.getSettings();
         this.indexCreatedVersion = indexSettings.getIndexVersionCreated();
         this.indexName = indexSettings.getIndex().getName();
+        this.indexMode = indexSettings.mode();
+
         List<String> fields = INDEX_SORT_FIELD_SETTING.get(settings);
+        if (indexMode.organizeIntoTimeSeries()) {
+            if (false == fields.isEmpty()) {
+                throw new IllegalArgumentException("Can't set [" + INDEX_SORT_FIELD_SETTING.getKey() + "] in time series mode");
+            }
+            if (INDEX_SORT_ORDER_SETTING.exists(settings)) {
+                throw new IllegalArgumentException("Can't set [" + INDEX_SORT_ORDER_SETTING.getKey() + "] in time series mode");
+            }
+            if (INDEX_SORT_MODE_SETTING.exists(settings)) {
+                throw new IllegalArgumentException("Can't set [" + INDEX_SORT_MODE_SETTING.getKey() + "] in time series mode");
+            }
+            if (INDEX_SORT_MISSING_SETTING.exists(settings)) {
+                throw new IllegalArgumentException("Can't set [" + INDEX_SORT_MISSING_SETTING.getKey() + "] in time series mode");
+            }
```

Comment:
Feels like it can be replaced with something like 
```
            for (Setting<?> setting: indexMode.unsupportedSettings()) {
                if (setting.exists(settings)) {
                    throw new IllegalArgumentException("Can't set [" + setting.getKey() + "] in time series mode");
                }
            }
```

or moved completely into `mode.checkSettings(settings)`

Reply:
:+1: